pydis_core/utils/error_handling/commands/manager.py

```diff
@@ -0,0 +1,61 @@
+from collections.abc import Callable
+from typing import NoReturn
+
+from discord import Interaction
+from discord.ext.commands import Context
+
+from pydis_core.utils.error_handling.commands import AbstractCommandErrorHandler
+from pydis_core.utils.logging import get_logger
+
+log = get_logger(__name__)
+
+
+class CommandErrorManager:
+    """A class that registers error handlers and handles all command related errors."""
+
+    def __init__(self, default: AbstractCommandErrorHandler, handlers: list[AbstractCommandErrorHandler] | None = None):
+        self.handlers = handlers or []
+        self.registered_handlers = set(type(handler).__name__ for handler in self.handlers)
+        self.default = default
+
+    async def handle_error(
+        self,
+        error: Exception,
+        context_or_interaction: Context | Interaction
+    ) -> None:
+        """
+        Handle a discord exception.
+
+        This works simply by iterating over the available handlers, and chooses the one capable of handling the error.
+
+        Instead of having some priority system to distinguish the execution order of handlers, we will
+        let the `should_handle_error` do the job.
+
+        The default needs to always be last, since the container can either be initialized with the handlers, or
+        we can add handlers using the `register_handler` function that we don't know when it will be invoked, if ever.
```

Comment:
I think implementation details should stay out of docstrings and instead be comments. Consumers/callers of the function likely do not need to know these details.

Reply:
Changed here bd62ce8